### ISSUE TITLE
"Fixed or Smaller (Bigger)" constraint for length and angle

### ISSUE THREAD

**Issue by Symbian9** [solvespace/solvespace#81]
Add constraint that give a way set parametre as ≤ or ≥ to inserted value.

**Comment by jwesthues** [comment]
In brief, SolveSpace's solver could be hacked to kind of support inequalities, but it was never made for that. Some earlier threads:

http://solvespace.com/forum.pl?action=viewthread&parent=211
http://solvespace.com/forum.pl?action=viewthread&parent=886

**Comment by Symbian9** [comment]
Also, some time its needed prevent to change vector (for example, "line could be enlarged only from start point in left direction, and not in reverse")

**Comment by Evil-Spirit** [comment]
@jwesthues, This actually was made and works fine. I even don't see any "hacks" in this solution, just the pure mathematics :)

**Comment by damian0815** [comment]
Are they any plans to bring this into the core? I would love this feature.

**Comment by whitequark** [comment]
No immediate plans, other tasks have a higher priority.

**Comment by Symbian9** [comment]
> This actually was made and works fine. I even don't see any "hacks" in this solution, just the pure mathematics :)

@jwesthues & @Evil-Spirit, so if it work in test, why not implemented yet?

**Comment by Evil-Spirit** [comment]
@jwesthues,
@whitequark,
@Symbian9,
Uff... I made it!
https://github.com/Evil-Spirit/solvespace-master/commit/6b7e99b068ac7daeef87d2aab808538fb1e38a4a

**Comment by Evil-Spirit** [comment]
`case Op::NORM: return a->PartialWrt(p)->Times(a->Abs()->Sgn());`

> This actually was made and works fine. I even don't see any "hacks" in this solution, just the pure mathematics :)

> I even don't see any "hacks"

> "hacks"

**Comment by Symbian9** [comment]
> Uff... I made it!

@Evil-Spirit, you are the man! Do not forget call @probonopd to this party too!

**Comment by Harvie** [comment]
Can you please merge this? Looks very nice!

https://github.com/KmolYuan/python-solvespace/issues/3

**Comment by whitequark** [comment]
Yeah, it's in my review queue. I did not forget about this patch though.

**Comment by Harvie** [comment]
BTW if we could just enter the inequality or range like this, it would be smooth:

![image](https://user-images.githubusercontent.com/276504/58156770-2a0da300-7c77-11e9-88a3-09e06b6b0af0.png)

(it already supports some kind of basic math like subtraction, so this can be nice addition)
